lucene/suggest/src/java/org/apache/lucene/search/suggest/document/TopSuggestDocs.java

```diff
@@ -116,19 +123,28 @@ public TopSuggestDocs(TotalHits totalHits, SuggestScoreDoc[] scoreDocs) {
    */
   public static TopSuggestDocs merge(int topN, TopSuggestDocs[] shardHits) {
     SuggestScoreDocPriorityQueue priorityQueue = new SuggestScoreDocPriorityQueue(topN);
+    boolean allComplete = true;
     for (TopSuggestDocs shardHit : shardHits) {
       for (SuggestScoreDoc scoreDoc : shardHit.scoreLookupDocs()) {
         if (scoreDoc == priorityQueue.insertWithOverflow(scoreDoc)) {
           break;
         }
       }
+      allComplete = allComplete && shardHit.isComplete;
     }
     SuggestScoreDoc[] topNResults = priorityQueue.getResults();
     if (topNResults.length > 0) {
-      return new TopSuggestDocs(new TotalHits(topNResults.length, TotalHits.Relation.EQUAL_TO), topNResults);
+      return new TopSuggestDocs(new TotalHits(topNResults.length, TotalHits.Relation.EQUAL_TO), topNResults,
+          allComplete);
     } else {
       return TopSuggestDocs.EMPTY;
     }
   }
 
+  /**
+   * returns true if we exhausted all possibilities to collect results
```

Comment:
nit: the flag indicates that the list of results is complete but we don't need to exhaust all possibilities to achieve this.  Maybe something like: `indicate if the list of results is complete or not, this might be <code>false</code> if the {@link TopNSearcher} rejected too many results.` ?

Reply:
Sounds good.